compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -632,13 +696,23 @@ static LogicalResult setRootConfig(
   if (isAArch64(*variantOp) && !isQuantized) {
     return setAArch64RootConfig(entryPointFn, contractionOp, flowTileSizes,
                                 workgroupTileSizes, vectorSize);
-  } else if (!disableMatmulPadPipeline &&
-             (isX86(*variantOp) || isRISCV(*variantOp))) {
+  }
+
+  TileSizesListType tileSizes = {flowTileSizes, workgroupTileSizes};
```

Comment:
I don't have a concrete idea yet, but I'll make this part smarter.

Reply:
Let me take a stab at rebasing on top of your PR. I might can simplify the logic a bit.